src/bigquery.ts

```diff
@@ -760,10 +820,16 @@ export class BigQuery extends common.Service {
     } else if (value instanceof Big) {
       typeName = 'NUMERIC';
     } else if (is.array(value)) {
-      return {
-        type: 'ARRAY',
-        arrayType: BigQuery.getType_(value[0]),
-      };
+      if (value.length === 0) {
+        throw new Error('Type must be provided for empty array.');
+      } else {
+        return {
+          type: 'ARRAY',
+          arrayType: BigQuery.getTypeDescriptorFromValue_(value[0]),
+        };
+      }
+    } else if (value === null) {
```

Comment:
To make branches like this less complex and more readable, I like to pull these to the top, and just do an `if` check that throws.  That way, you need fewer branched `else` clauses.